### PR TITLE
.Net: Address IL2067 and IL3051 AOT warnings in Abstractions project

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation, Context and Description
This PR addresses the last two IL2067 and IL3051 AOT warnings in the Abstractions project. It also adds the testing app that uses   .NET’s AOT compiler to analyze SK projects and ensure all AOT related warnings are found. You can find more details here: - https://devblogs.microsoft.com/dotnet/creating-aot-compatible-libraries/#analyzing-.net-libraries.

### CODE REVIEW

File: dotnet/src/SemanticKernel.Abstractions/Services/EmptyServiceProvider.cs [hunk 1]
```diff
@@ -32,9 +33,29 @@ public object GetRequiredKeyedService(Type serviceType, object? serviceKey) =>
...
+    private static Array CreateArray(Type elementType, int length)
+    {
+        if (VerifyAotCompatibility && elementType.IsValueType)
+        {
+            // NativeAOT apps are not able to make Enumerable of ValueType services
+            // since there is no guarantee the ValueType[] code has been generated.
+            throw new InvalidOperationException($"Unable to create an Enumerable service of type '{elementType}' because it is a ValueType. Native code to support creating Enumerable services might not be available with native AOT.");
+        }
+
+        return Array.CreateInstance(elementType, length);
```
Comment (SergeyMenshykh) [inline dotnet/src/SemanticKernel.Abstractions/Services/EmptyServiceProvider.cs:52]: anchor

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/Aot/AotCompatibility.TestApp/AotCompatibility.TestApp.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (markwallace-microsoft) [inline dotnet/src/Aot/AotCompatibility.TestApp/AotCompatibility.TestApp.csproj:1]: Should this be in the `samples\Demos` folder?

**Review by markwallace-microsoft — APPROVED** [review]

**Review by RogerBarreto — APPROVED** [review]
Move the TestApp to the `Demo`, LGTM

File: dotnet/src/Aot/AotCompatibility.TestApp/AotCompatibility.TestApp.csproj (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/Aot/AotCompatibility.TestApp/AotCompatibility.TestApp.csproj:1]: The test application serves two purposes: uncovering all AOT-related warnings and running E2E scenarios that use the SK AOT APIs to test them. We can certainly consider a third purpose: demonstrating how certain scenarios can be configured in AOT.

Comment (SergeyMenshykh) [reply dotnet/src/Aot/AotCompatibility.TestApp/AotCompatibility.TestApp.csproj]: Done

**Pull request merged**